src/main/java/com/google/research/bleth/simulator/BoardStateFactory.java

```diff
@@ -57,7 +57,7 @@ public static BoardState create(Board board, String simulationId, int round) {
         int colNum = board.getColNum();
         ArrayTable<Integer, Integer, ArrayList<String>> boardStateTable = createEmptyTable(rowNum, colNum);
         Multimap<Location, IAgent> agentsOnBoard = board.agentsOnBoard();
-        for (Location location : agentsOnBoard.keys()) {
+        for (Location location : agentsOnBoard.keySet()) {
```

Comment:
(I made the same comment on Omer's similar change)

Nit: this seems like an independent small fix. Prefer to submit those in a separate small pull-request, as they're trivial to approve (and - conversely - when combined into an unrelated PR, e.g. here, this is just confusing).

Reply:
It is related to the PR; I found this bug because of the tests. But I'll open new PR next time.